hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientFailoverConfigBuilder.java

```diff
@@ -72,17 +73,40 @@ public XmlClientFailoverConfigBuilder(InputStream in) {
     }
 
     /**
-     * Loads the client config using the following resolution mechanism:
+     * Loads the client failover config using the following resolution mechanism:
      * <ol>
      * <li>first it checks if a system property 'hazelcast.client.failover.config' is set. If it exist and it begins with
      * 'classpath:', then a classpath resource is loaded. Else it will assume it is a file reference</li>
      * <li>it checks if a hazelcast-client-failover.xml is available in the working dir</li>
      * <li>it checks if a hazelcast-client-failover.xml is available on the classpath</li>
-     * <li>if none available build throws HazelcastException </li>
      * </ol>
+     *
+     * @throws HazelcastException if no failover configuration is found
```

Comment:
Why not `InvalidConfigurationException`?

Reply:
I _guess_ it's because there is no config found and not that an invalid configuration is provided. I added this to the javadoc because this exception is thrown from the constructor. @sancar, you added the line throwing this exception. Could you please clarify if that's the case? Could it be a `ConfigurationException` or `IllegalConfigurationException` instead?

Reply:
Yes, this is not related to this prd. 
My reasoning was as follows:
`InvalidConfigurationException`  is used when an invalid value is provided inside the config. 
This case looks different. No config is found to load. 
And since `HazelcastException` is used other places like ClasspathXmlConfig, it seemed more suitable to me.